runtime/lua/vim/treesitter/query.lua

```diff
@@ -175,11 +173,22 @@ function M.get_node_text(node, source)
   local end_row, end_col, end_byte = node:end_()
 
   if type(source) == "number" then
-    if start_row ~= end_row then
-      return nil
+    local lines
+    if end_col == 0 then
+      lines = a.nvim_buf_get_lines(source, start_row, end_row, true)
+      end_col = #lines[#lines]
+    else
+      lines = a.nvim_buf_get_lines(source, start_row, end_row + 1, true)
     end
-    local line = a.nvim_buf_get_lines(source, start_row, start_row+1, true)[1]
-    return string.sub(line, start_col+1, end_col)
+    lines[1] = string.sub(lines[1], start_col + 1)
+
+    local end_index = end_col
+    if #lines == 1 then
+      end_index = end_col - start_col
+    end
+    lines[#lines] = string.sub(lines[#lines], 1, end_index)
+
+    return table.concat(lines, "\n")
```

Comment:
Would it make sense to have a `nvim_buf_get_text` and use that here?

Reply:
Turns out there is already a PR for it - https://github.com/neovim/neovim/pull/15181   once it is merged I can update the PR here to use it.